webview-ui/src/components/settings/ApiOptions.tsx

```diff
@@ -1105,7 +1101,7 @@ const ApiOptions = ({ apiErrorMessage, modelIdErrorMessage, fromWelcomeView }: A
 						value={apiConfiguration?.lmStudioBaseUrl || ""}
 						style={{ width: "100%" }}
 						type="url"
-						onBlur={handleInputChange("lmStudioBaseUrl")}
+						onInput={handleInputChange("lmStudioBaseUrl")}
```

Comment:
Does this need to change? I've been trying to get away from the onInput updates.

Reply:
This is to avoid useless interval calls. The model list is loaded only when the user switches the provider or changes the URL (on-demand loading to avoid UI refresh every two seconds).
Don't worry about triggering a large number of loading requests. This part has been debounced. Loading will only be triggered after the user stops typing for 250ms.
If this modification is revoked, the model list will not be updated until the input box loses focus, and the user cannot perceive whether the input is correct during typing.

Reply:
Got it, that makes sense for loading the model list. Isn't there also something where state change messages get sent back and forth from the webview on every keystroke as well though? The intention of my change from onInput to onBlur was to avoid that, though I hear you about not getting immediate feedback to the users.

Reply:
I have removed postStateToWebview from upsertApiConfiguration so that it only saves the configuration and does not update the UI in reverse again.
However, this breaks the function of automatically switching to a new profile when adding a profile. Next, I will split it into saveApiConfiguration to specifically provide a save function without forced UI updates.

Reply:
Ahh I see! Should we revert my changes in https://github.com/RooVetGit/Roo-Code/pull/889 then?

Reply:
Currently, only the fields that trigger model loading need onInput events, and other fields can keep onBlur.
Since the save process is asynchronous, in order to avoid old values ​​overwriting new values, the save frequency should be reduced.
Ideally, all input saves should use the onBlur event, and onInput is only used to update the model list. I will submit another modification later to solve potential asynchronous concurrency issues.

Reply:
I have changed the code but it seems that the unit test of the current main branch does not pass.

Reply:
BTW, #889 is fine, no need to revert it.